src/content/blog/contributing-to-a-create-typescript-app-repository/index.mdx

```diff
@@ -0,0 +1,353 @@
+<LabeledVideo
+	alt="Eleanor from 'The Good Place' happily saying 'come on in'."
+	description="No matter your experience level, I'm excited for you to contribute in open source! 💜"
+	src="/images/blog/the-good-place-come-on-in.webm"
+/>
+
+## Prerequisites
+
+You'll need to install the following two tools on whatever computer/environment you want to develop on:
+
+- [Node.js](https://nodejs.org) to run JavaScript
+- [pnpm](https://pnpm.io): A alternative to Node's `npm` package manager that is generally much faster
+
+I recommend [VS Code](https://code.visualstudio.com) on your own computer because that's what I personally use and can help with the most.
+This blog post assumes you're using VS Code to keep the explanations straightforward.
+But if you feel more comfortable with a different setup, that's great too.
+Just make sure you're following the equivalent steps for your editor.
+
+My repositories also include a lot of additional tooling that you'll likely need to work with.
+You don't have to know any of the additional tooling already, but you should be able to read documentation, put forth a real effort to understand what the docs saying, and ask specific questions if you get stuck.
+
+Lastly, read the repository's contributing guide (`.github/CONTRIBUTING.md`) and development guide (`.github/DEVELOPMENT.md`).
+They'll tell you each of the commands used in that repository.
+
+> 💡 Whenever you use a tool for the first time, **read the tool's documentation**.
+> It will save you _so much time_ and make working much less confusing.
+
+### Git and GitHub Familiarity
+
+You don't need to be confident in Git or GitHub to contribute.
+I've included common steps to take in this guide.
+But it's helpful to understand vaguely what a "repository" is and what the difference between Git and GitHub are.
+At the very least, go through GitHub's _[Hello World](https://docs.github.com/en/get-started/quickstart/hello-world)_ and _[Contributing to Projects](https://docs.github.com/en/get-started/quickstart/contributing-to-projects)_ guides.
+
+## Finding an Issue
+
+The first step for almost all contributions is to find an open issue on the repository's issues page that is:
+
+- Marked as `accepting prs`: indicating it's something I'll accept a pull request for
+- Does not have an assignee: meaning no maintainer is planning on working on it
+
+Unless the change you'd like to make is an exceedingly small typo, please don't skip this step.
+Respecting the issue process gives several benefits:
+
+- It gives a repository's maintainer(s) the opportunity to provide useful advice
+- It makes sure contributors don't spend hours of work that won't be accepted
+- It keeps a discoverable, organized history of why changes were made
+
+I recommend finding an issue labeled `good first issue` if you've never worked in a particular repository before.
+Those issues are generally smaller, more self-contained ones that the maintainer(s) think should be good for first-timers.
```

Comment:
A note from Michael Faith on Mastodon (https://bsky.app/profile/michael.faith/post/3kbo6hk4bdw2h):

> One recommendation: in the "Finding an Issue" section, it might also be good to consider people that want/need to log an issue themselves, that they then contribute a solution for.  It mostly focuses on people looking for existing issues.


Agreed, thanks!

Reply:
Looks good